doc/neomake.txt

```diff
@@ -1049,4 +1049,34 @@ Here are some tips to develop the 'errorformat' setting for makers:
    display the raw data.
 5. Pay attention to the "`valid`" property of entries.
 
+                                               *neomake-faq-local-executables*
+                                               *neomake-faq-eslint*
+
+6.4 How to prefer local over global executables?~
+
+Some programs such as eslint are typically found in both a global and a local
+location. The global location is usually included in your shell path, whereas
+the path to the locally installed executable for the current project is often
+not included in your path (run `$ echo $PATH` in your shell to see the current
+path)
+
+Neomake itself has no special logic to determine the correct path to any
+executable, meaning if for example eslint is only available globally, that's
+what Neomake will use. This often breaks eslint, since the plugins installed
+in your local project are not also installed globally.
+
+There are several ways of addressing this issue:
+
+- Add `./node_modules/.bin` to your shell path, *before* the entry for the
```

Comment:
This should use an absolute path likely (since Neomake might change the cwd before running the maker).
(This might have been the problem with your previous comment in https://github.com/neomake/neomake/issues/2340)

Reply:
Good point, in my current `config.fish` I'm actually using `$PWD/node_modules/.bin`

Reply:
Yeah.. it should be probably set when `cd`ing to your project (e.g. via https://github.com/Tarrasch/zsh-autoenv).  Also https://github.com/embear/vim-localvimrc/ could be used to set this only in Vim/Neovim then.
Another option might be to use Neomake's hooks (autocommands to do this).

I am not sure if relative `$PATH` entries are meant to work in general.
It could make sense to use `exepath()` to get the full path of the executable before doing any `cd` for the job.  I'll create a new issue to track this.

Reply:
I updated the description. I'm not really happy with the `$PATH` part to be honest since it has a lot of caveats, a lot of ways of achieving the goal and it depends on which shell the user uses. I therefore moved this suggestion to the bottom of the list.